src/gt4py/next/iterator/ir.py

```diff
@@ -186,6 +186,10 @@ def _output_validator(self: datamodels.DataModelTP, attribute: datamodels.Attrib
     "cast_",
 }
 
+GTIR_BUILTINS = {
```

Comment:
```suggestion
GTIR_BUILTINS = {
```
Does `GTIR` really make sense here? After all everything will be part of GTIR and you don't really make use of the fact that this is in a seperate list. What I did in my PR was to only use the new builtins in the `Program` so that they can not accidentally be used in a Fencil.

```python
GTIR_BUILTINS = {
    *BUILTINS,
    "apply_stencil"
}

class Program(Node, ValidatedSymbolTableTrait):
    id: Coerced[SymbolName]
    function_definitions: List[FunctionDefinition]
    params: List[Sym]
    declarations: List[Temporary]
    body: List[Stmt]

    _NODE_SYMBOLS_: ClassVar[List[Sym]] = [Sym(id=name) for name in GTIR_BUILTINS]
```

Reply:
makes sense.